src/components/MultiHopTrade/hooks/useGetTradeQuotes/useGetTradeQuotes.tsx

```diff
@@ -283,18 +298,21 @@ export const useGetTradeQuotes = () => {
     return () => clearInterval(interval)
   }, [])
 
-  // NOTE: we're using currentData here, not data, see https://redux-toolkit.js.org/rtk-query/usage/conditional-fetching
-  // This ensures we never return cached data, if skip has been set after the initial query load
-  const { currentData } = useGetTradeQuoteQuery(tradeQuoteInput, {
+  useGetTradeQuoteQuery(tradeQuoteInput, {
     skip: !shouldRefetchTradeQuotes,
     pollingInterval: hasFocus ? GET_TRADE_QUOTE_POLLING_INTERVAL : undefined,
     /*
-      If we don't refresh on arg change might select a cached result with an old "started_at" timestamp
-      We can remove refetchOnMountOrArgChange if we want to make better use of the cache, and we have a better way to select from the cache.
-     */
+    If we don't refresh on arg change might select a cached result with an old "started_at" timestamp
+    We can remove refetchOnMountOrArgChange if we want to make better use of the cache, and we have a better way to select from the cache.
+    */
     refetchOnMountOrArgChange: true,
   })
 
+  // NOTE: we're using currentData here, not data, see https://redux-toolkit.js.org/rtk-query/usage/conditional-fetching
+  // This ensures we never return cached data, if skip has been set after the initial query load
+  // currentData is always undefined when skip === true, so we have to access it like so:
+  const { currentData } = swappersApi.endpoints.getTradeQuote.useQueryState(tradeQuoteInput)
```

Comment:
note the above:
```
currentData is always undefined when skip === true, so we have to access it like so
```

Reply:
If I understand the intent correctly here, we're introspecting the `queryState` without the `UseQueryStateOptions`, so we can get the state for the last succesful query? 
If that's indeed the intent, we could simply destructure `data` instead of `currentData`, which would make things simpler to grasp

Reply:
the key is that we dont want to be pushing up cached responses to mixpanel, so ideally we keep it as-is unless `data` wont change across equivalent api requests?

i'll follow up with you async so we can get this in and solve this api issue separately